risc0/zkvm/src/prove/external.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    fs::File,
+    io::Write,
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
+
+use anyhow::{anyhow, bail, Result};
+use risc0_binfmt::MemoryImage;
+use tempfile::tempdir;
+
+use super::Prover;
+use crate::{ExecutorEnv, Receipt, Segment, SegmentReceipt, Session, VerifierContext};
+
+/// An implementation of a [Prover] that runs proof workloads via an external
+/// `r0vm` process.
+pub struct ExternalProver {
+    name: String,
+    r0vm_path: PathBuf,
+}
+
+impl ExternalProver {
+    /// Construct an [ExternalProver].
+    pub fn new<P: AsRef<Path>>(name: &str, r0vm_path: P) -> Self {
```

Comment:
Here, I think we could add a hash function name as a parameter to `new`. Similar to the LocalProver, we can have the caller/user request a hash function to be used. When creating an instance of the External prover, we could call r0vm to see a list of supported hash functions. If the requested hash function doesn't exist, we can do  `bail("hash function {} is not supported by r0vm. Supported hash functions {}", requested_hash, supported_hash_list)`.

Reply:
done